gen3-client/jwt/functions.go

```diff
@@ -184,8 +180,54 @@ func (f *Functions) DoRequestWithSignedHeader(profile string, configFileType str
 		f.Config.UpdateConfigFile(cred, []byte(content), cred.APIEndpoint, configPath, profile)
 
 		resp, err = f.Request.MakeARequest(method, apiEndpoint, cred.AccessKey, contentType, bytes.NewBuffer(bodyBytes))
-		msg, err = f.ParseFenceURLResponse(resp)
-		return msg.Url, msg.GUID, err
 	}
+
+	return prefixEndPoint, resp, nil
+}
+
+func (f *Functions) DoRequestWithSignedHeader(profile string, configFileType string, endpointPostPrefix string, contentType string, bodyBytes []byte) (string, string, error) {
+	/*
+	   Do request with signed header. User may have more than one profile and use a profile to make a request
+	*/
+	var err error
+	var msg JsonMessage
+
+	_, resp, err := f.GetResponse(profile, configFileType, endpointPostPrefix, contentType, bodyBytes)
+	if err != nil {
+		return "", "", err
+	}
+
+	msg, err = f.ParseFenceURLResponse(resp)
+	return msg.Url, msg.GUID, err
+
+	panic("Unexpected case")
+}
+
+func (f *Functions) CheckPrivileges(profile string, configFileType string, endpointPostPrefix string, contentType string, bodyBytes []byte) (string, map[string]interface{}, error) {
+	/*
+	   Return user privileges from specified profile
+	*/
+	var err error
+	var data map[string]interface{}
+
+	host, resp, err := f.GetResponse(profile, configFileType, endpointPostPrefix, contentType, bodyBytes)
+	if err != nil {
+		return "", nil, err
+	}
+
+	str := ResponseToString(resp)
+
+	err = json.Unmarshal([]byte(str), &data)
+	if err != nil {
+		panic(err)
```

Comment:
In general we should discourage to use `panic` unless there is a `defer-panic-recover` chain. Instead you can just use `log.Fatal` or `log.Print`.

I know there are still some `panic` leftovers in the current code. But they are going to be replaced eventually (and hopefully) so let's not introduce new members to them 😄 

Also, for this `err`, you could just return it and let the caller of this function to deal with it, just like all other errs in this function, to keep a consistent behavior

Reply:
Good to know, thanks! Fixed in the commit [33dfedf](https://github.com/uc-cdis/cdis-data-client/pull/49/commits/33dfedfc97fbef84fe5d7a2465d075d60aa6d4dd)